libs/community/langchain_community/embeddings/vertexai.py

```diff
@@ -20,6 +24,13 @@ def validate_environment(cls, values: Dict) -> Dict:
             from vertexai.language_models import TextEmbeddingModel
         except ImportError:
             raise_vertex_import_error()
+        if values["model_name"] == "textembedding-gecko-default":
+            logger.warning(
+                "Model_name will become a required arg for VertexAIEmbeddings "
+                "starting from Feb-01-2024. Currently the default is set to "
+                "textembedding-gecko@001"
+            )
+            values["model_name"] = "textembedding-gecko@001"
```

Comment:
what's the difference between textembedding-gecko and textembedding-gecko@001? does the former track latest version

Reply:
`textembedding-gecko@001` points to the specific version while `textembedding-gecko` to the default one that can be changed.

Reply:
if google is exposing a default version in the API is it wrong to default to that here, too?

Reply:
The Embeddings API actually uses the model name as a path parameter, which makes it a required input: https://cloud.google.com/vertex-ai/docs/generative-ai/embeddings/get-text-embeddings#supported_models

We want to make the model name required input here as well, so we are starting with this small change to pin the version and warn of the upcoming break. Pinning the version is also consistent with the best practices called out in that documentation and how OpenAIs model number is pinned